src/utils/cardsDetailsUtils.ts

```diff
@@ -0,0 +1,123 @@
+import Character from '../entity/Character';
+import { MysqlDataSource } from '../config/database';
+import { CharacterComics } from '../entity/CharacterComics';
+import Comic from '../entity/Comic';
+import Series from '../entity/Series';
+import Event from '../entity/Event';
+import { CharacterSeries } from '../entity/CharacterSeries';
+import { CharacterEvents } from '../entity/CharacterEvents';
+import Story from '../entity/Story';
+import { CharacterStories } from '../entity/CharacterStories';
+
+/**
+ * 
+ * @param {Character} character - Personagem sobre o qual se quer obter os quadrinhos relacionados
+ * @returns {Array<Object>} - Retorna um array de objetos JSON, no formato {id, description, thumb}
+ */
+export async function getComicsByCharacter(
+  character: Character
+): Promise<Comic[]> {
+  const characterComicsRepository =
+    MysqlDataSource.getRepository(CharacterComics);
+
+  const characterComics = await characterComicsRepository.find({
+    where: { character: character },
+    relations: ['comic']
+  });
+
+  const comics = characterComics.map((characterComic) => characterComic.comic);
+
+  const formattedComics = selectFewValues(comics);
+
+  return formattedComics;
+}
+
+/**
+ * 
+ * @param {Character} character - Personagem sobre o qual se quer obter as series relacionadas
+ * @returns {Array<Object>} - Retorna um array de objetos JSON, no formato {id, description, thumb}
+ */
+export async function getSeriesByCharacter(
+  character: Character
+): Promise<Series[]> {
+  const characterSeriesRepository =
+    MysqlDataSource.getRepository(CharacterSeries);
+
+  const characterSeries = await characterSeriesRepository.find({
+    where: { character: character },
+    relations: ['series']
+  });
+
+  const series = characterSeries.map(
+    (characterSeries) => characterSeries.series
+  );
+
+  const formattedSeries = selectFewValues(series);
+
+  return formattedSeries;
+}
+
+/**
+ * 
+ * @param {Character} character - Personagem sobre o qual se quer obter os eventos relacionados
+ * @returns {Array<Object>} - Retorna um array de objetos JSON, no formato {id, description, thumb}
```

Comment:
Aqui o ideal e tipar como Promise<Event[]> mesmo, isso também vale para os outros métodos.

Reply:
Você pode criar interfaces:

Reply:
https://cynoteck.com/pt/blog-post/interfaces-in-typescript/

Reply:
Feito, criei a interface formattedCards e alterei a documentação.